code/modules/food_and_drinks/kitchen_machinery/grill_new.dm

```diff
@@ -43,3 +43,36 @@
 	for(var/obj/item/weapon/stock_parts/micro_laser/M in component_parts)
 		E += M.rating
 	efficiency = round((E/2), 1) // There's 2 lasers, so halve the effect on the efficiency to keep it balanced
+
+/obj/machinery/kitchen_machine/grill/special_attack(obj/item/weapon/grab/G, mob/user)
+	if(istype(G.affecting, /mob/living/carbon/human))
+		if(G.state < GRAB_AGGRESSIVE)
+			to_chat(user, "<span class='warning'>You need a better grip to do that!</span>")
+			return 1
+		var/mob/living/carbon/human/C = G.affecting
+		var/obj/item/organ/external/head/head = C.get_organ("head")
+		var/obj/item/organ/external/chest/chest = C.get_organ("chest")
+		var/obj/item/organ/external/arm/l_arm = C.get_organ("l_arm")
+		var/obj/item/organ/external/arm/right/r_arm = C.get_organ("r_arm")
```

Comment:
This is a whole lot of checking and complexity that probably really isn't necessary, when you can just....